src/clients/gfycat/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { Either } from "monet";
+import { map, pipe } from "ramda";
+import { GfycatApiResponse, GfycatApiItem } from "./gfycat-api";
+import { Gif } from "../../Gif";
+import {
+  convertEithersToMaybes,
+  getGifsFromMaybes,
+} from "../../utils/monad-utils";
+
+const getResultsFromResponse = (
+  response: GfycatApiResponse
+): Array<GfycatApiItem> => response.gfycats || [];
+
+const convertResultsToEithers = map((item: GfycatApiItem) =>
+  Either.fromTry(
+    () =>
+      new Gif(
+        item?.gifUrl || item?.max5mbGif || item?.max2mbGif || item?.max1mbGif,
+        item?.max1mbGif || item?.max2mbGif || item?.max5mbGif || item?.gifUrl
```

Comment:
I changed these lines since we were previously using the non-existant `poster_url` (it's actually posterUrl).

I removed the posterUrl completely though as that returns a static image. Since the other Strategies return low-quality Gifs in the thumbnailUrl field, I did the same here.

The long `||` chain is required to handle the fact that Gfycat can return objects with sometimes non-existant fields.

Reply:
This should have been caught in the tests. I'll update them seperately to keep this commit clean.

Reply:
Thanks for the quick feedback on all this! Glad the process was nice and smooth 👍